specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,306 @@
+- [`Components\NimbleThemeProvider.razor`](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b#diff-935610f630ad061236774e447badf29f08dd1467f6c4c6e45ead7214641afe6d): Add code/markup for labels: i.e. `label-number-field-increment="@LabelNumberFieldIncrement"` in the Razor file, `[Parameter] public string? LabelNumberFieldIncrement { get; set; }` in the C# code. We may need to fully codegen the Razor file to add the attribute bindings.
+
+Once clients add a reference to the NuGet, `NimbleResources\NimbleLabels.resx` will appear in their project as a link (to that file in the NuGet). At that point, the remaining process is somewhat convoluted, and would need documentation. The prototype branch has done some of this in `Demo.Shared` and `Demo.Server`:
+- Browse to that resx file on disk, and manually copy it into the app project directory, in `Resources\`
+- Add it to the Blazor project (as Build Action = Embedded Resource). Then, open it in the resx editor, and set Access Modifier = Public or Internal (to generate the `.designer.cs` file)
+- To translate/localize the resources, clone the resx file (e.g. `NimbleLabels.fr.resx`) and translate the values in it. The app also needs to follow the [Microsoft Blazor localization docs](https://learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-7.0&pivots=server) in terms of opting into localization, configuring the supported locales, etc.
+- To consume the resources, in the Razor file using the NimbleThemeProvider, add:
+```cs
+[Inject]
+internal IStringLocalizer<NimbleLabels>? LabelProvider { get; set; }
+```
+Then bind to the theme provider labels in the Razor template:
+```xml
+<NimbleThemeProvider Theme="Theme" LabelNumberFieldDecrement="@LabelProvider!["labelNumberFieldDecrement"]" LabelNumberFieldIncrement="@LabelProvider!["labelNumberFieldIncrement"]">
+```
+
+The process above isn't great. When updated versions of Nimble Blazor are released, if they have new labels, the consuming app would need to repeat the process of manually copying the resx, and translating it. They'd also need to manually add the new label properties in their Razor file using the NimbleThemeProvider.
+
+*Possible Alternatives:*
+- Add an `IStringLocalizer` property to the `NimbleThemeProvider` Razor component, and codegen the `Label*` properties to use it if it's set. This would improve the process of telling Nimble about the localized resources, but wouldn't change the process of manually copying the resx to start with.
+- Define an MSBuild task that copies the resx file, similar to what's outlined in [this GitHub comment](https://github.com/ni/nimble/issues/558#issuecomment-1129279985). Not much better / still a manual process.
+
+## Open Issues
+
+- Naming
+   - Do we like having `i18n` in the element names, or should we pick something else? Once camel-cased it also looks strange, i.e. `NimbleI18nCore.razor`
```

Comment:
Some ideas:
- NimbleIntlCore, nimble-intl-core, nimble-intl-table (like browser Intl api)
- NimbleLangCore, nimble-lang-core, nimble-lang-table (like browser lang attribute)
- NimbleLabelsCore, nimble-labels-core, nimble-labels-table (i.e. lean in on the tokens being configured)
- NimbleLabelProvider, nimble-label-provider, nimble-table-label-provider (lean in on the design token provider aspect)

I think I kinda like the last one. Good symmetry to the theme provider and it says what it is providing, i.e. the label tokens.

Reply:
Updated to nimble-label-provider-core, nimble-label-provider-table